pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -421,20 +477,30 @@ class ModelInterpreter implements Serializable {
         parameters
     }
 
+    /**
+     * Legacy version to pass the root tools in, rather than directly passing in a tools. Only relevant for in-progress
+     * runs.
+     * TODO: Delete in 1.4? Or maybe just nuke now.
```

Comment:
Don't know either. I don't remember how it behaves when restoring an old running pipeline with this new code. If the old code is what is run or not.